File: Application/warframe-progress-tracker/src/app/melee-weapon/melee-weapon.ts
export interface MeleeWeapon {
   id: number;
   name: string;
   masteryRequired: number;
   type: string;

   attackSpeed: number;
   rangeMetres: number;
   comboSeconds: number;
   blockDegrees: number;
   followThrough: number;
   disposition: number;

   attacks: object[];

   augments: number[];
   polarities: string;
   stancePolarity: string;
   stances: number[];
}

File: Server/WarframeProgressTrackerApi/WarframeProgressTrackerApi/Models/MeleeWeapon.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace WarframeProgressTrackerApi.Models {
    public class MeleeWeapon : WarframeItem {
        public int MasteryRankRequired { get; set; }
        public string WeaponType { get; set; }
        public float RangeMetres { get; set; }
        public float AttackSpeed { get; set; }
        public int BlockAngleDegrees { get; set; }
        public float ComboSeconds { get; set; }
        public float FollowThrough { get; set; }
        public float Disposition { get; set; }
        public string HowToGet { get; set; } = "";
    }
}
